### ISSUE TITLE
Add this to readme, for easy copy and paste and install, for those who use use-package and straight.

### ISSUE THREAD

**Issue by raveensrk** [jdtsmith/speedrect#1]
```elisp
(use-package speedrect
  :straight (speedrect :type git :host github :repo "jdtsmith/speedrect")
  )
```

**Comment by jdtsmith** [comment]
Fixed in 9bd1354.

**Comment by rdiaz02** [comment]
If one has `(setq use-package-always-defer t)`  the instructions for `use-package` (I don't know about straight) do not work. This works

```elisp
(use-package speedrect
  :ensure t
  :init (speedrect-mode 1))
```

Maybe it is worth mentioning this in the README?

**Comment by jdtsmith** [comment]
Did you first install it from ELPA?

**Comment by rdiaz02** [comment]
Yes, I first installed from ELPA. A reproducible example is:

1. Install from ELPA.
2. `emacs -Q`
3. `M-x package-initialize`
4. In the scratch buffer, evaluate `(setq use-package-always-defer t)`
5. In the scratch buffer, copy and evaluate
```
(use-package speedrect
  :config (speedrect-mode))
```
6. Type `C-x SPC`. SpeedRect is not active (e.g., the "SpeedRect: [?] for help" is not shown in the minibuffer).

And, from my understanding of what `defer` does with `use-package` what I am seeing is expected. `(setq use-package-always-defer t)` is setting deferred loading, so the package will not be loaded unless explicitly  called by trigger keywords, which are (from https://www.gnu.org/software/emacs/manual/html_node/use-package/Deferring-loading.html)   :commands, :bind, :bind*, :bind-keymap, :bind-keymap*, :mode, :interpreter and :hook.  But `(speedrect-mode)` is in the `:config` section, and from https://www.gnu.org/software/emacs/manual/html_node/use-package/Config-keyword.html   "The :config section is evaluated after the package has been loaded. If the package is loaded immediately, this happens immediately after that, but if loading is done lazily (see [Loading Packages](https://www.gnu.org/software/emacs/manual/html_node/use-package/Loading-Packages.html)), this is deferred until after the package has been loaded."  Therefore,  when using deferred loading, `(speedrect-mode)` will not be evaluated if it is in the `:config` section.


Of course, after step 6. above, one can activate the package manually just fine: `M-x speedrect-mode`.

**Comment by jdtsmith** [comment]
Always-defer does what it says. Since there are no top level bindings in place (or needed), you need to countermand it with `:demand t`. Can you give that a try and let me know how that works?

**Comment by rdiaz02** [comment]
Yes, just checked and adding `:demand t`  works.   

But (just my ignorance here): why is `:demand t`  better than `:init (speedrect-mode 1)`? `:demand t`, as per documentation, is "forc[ing] immediately loading anyway". Aren't the two doing exactly the same thing?

(What I like about `:init` is that, in my mental model of what `use-package` is doing, this is saying "regardless of anything else, evaluate this, period". But the `:config` + `demand` is, I think, saying "the config line will be evaluated when loaded; and since we wrote `:demand t` too, it will be loaded immediately, and thus the config line evaluated"; but isn't it like adding an extra layer of indirection?)

**Comment by jdtsmith** [comment]
It's easy to check, just `M-x pp-macroexpand-last-exp` at the end of both forms (maybe `C-x x n` to clone the first expansion buffer, then `M-x diff-buffers` after you expand the second).  You can see this way that the difference is mostly in _reporting_: warning, elapsed load time, etc. 


```diff
diff -u /var/folders/by/19mt78cj63v5_tslqh826g7w0000gn/T/buffer-content-8hNyiR /var/folders/by/19mt78cj63v5_tslqh826g7w0000gn/T/buffer-content-qz4Bp2
--- /var/folders/by/19mt78cj63v5_tslqh826g7w0000gn/T/buffer-content-8hNyiR	2024-12-19 10:34:19
+++ /var/folders/by/19mt78cj63v5_tslqh826g7w0000gn/T/buffer-content-qz4Bp2	2024-12-19 10:34:19
@@ -1,5 +1,5 @@
 (progn
-  (defvar use-package--warning123
+  (defvar use-package--warning120
     #'(lambda
 	(keyword err)
 	(let
@@ -8,9 +8,47 @@
 		      (error-message-string err))))
 	  (display-warning 'use-package msg :error))))
   (condition-case-unless-debug err
-      (condition-case-unless-debug err
-	  (speedrect-mode 1)
-	(error
-	 (funcall use-package--warning123 :init err)))
+      (let
+	  ((now
+	    (current-time)))
+	(message "%s..." "Loading package speedrect")
+	(prog1
+	    (if
+		(not
+		 (require 'speedrect nil t))
+		(display-warning 'use-package
+				 (format "Cannot load %s" 'speedrect)
+				 :error)
+	      (let
+		  ((now
+		    (current-time)))
+		(message "%s..." "Configuring package speedrect")
+		(prog1
+		    (condition-case-unless-debug err
+			(progn
+			  (speedrect-mode)
+			  t)
+		      (error
+		       (funcall use-package--warning120 :config err)))
+		  (let
+		      ((elapsed
+			(float-time
+			 (time-subtract
+			  (current-time)
+			  now))))
+		    (if
+			(> elapsed 0.1)
+			(message "%s...done (%.3fs)" "Configuring package speedrect" elapsed)
+		      (message "%s...done" "Configuring package speedrect"))))))
+	  (let
+	      ((elapsed
+		(float-time
+		 (time-subtract
+		  (current-time)
+		  now))))
+	    (if
+		(> elapsed 0.1)
+		(message "%s...done (%.3fs)" "Loading package speedrect" elapsed)
+	      (message "%s...done" "Loading package speedrect")))))
     (error
-     (funcall use-package--warning123 :catch err))))
+     (funcall use-package--warning120 :catch err))))
```

**Comment by rdiaz02** [comment]
Thanks a lot for the details!
